### DISCUSSION TITLE
Drone starts spinning after landing three times.

### DISCUSSION THREAD

**Discussion by noahra** [USC-ACTLab/crazyswarm#746] [Q&A]
I am using the Crazyflie drone and steering it with the Crazyswarm commands. After landing three time and then taking off, the drone starts spinning uncontrollably. What can I do in order to mitigate this?

**Comment by whoenig** [comment]
What system do you use for tracking? Sounds like a potential issue with the state estimation.

**Reply by noahra** [reply]
I am using a Mocap OptiTrack system and a single marker on my drone.

**Reply by noahra** [reply]
I am using ROS Noetic if that is important. Here is my hover_swarm.launch file. Estimator is set to Kalman it seems like.

```
<?xml version="1.0"?>
<launch>
  <arg name="joy_dev" default="/dev/input/js0" />

  <rosparam command="load" file="$(find crazyswarm)/launch/crazyflieTypes.yaml" />
  <rosparam command="load" file="$(find crazyswarm)/launch/crazyflies.yaml" />

  <node pkg="crazyswarm" type="crazyswarm_server" name="crazyswarm_server" output="screen" >
    <rosparam>
      # Logging configuration (Use enable_logging to actually enable logging)
      genericLogTopics: ["log1"]
      genericLogTopicFrequencies: [10]
      genericLogTopic_log1_Variables: ["stateEstimate.x", "ctrltarget.x"]
      # firmware parameters for all drones (use crazyflieTypes.yaml to set per type, or
      # allCrazyflies.yaml to set per drone)
      firmwareParams:
        commander:
          enHighLevel: 1
        stabilizer:
          estimator: 2 # 1: complementary, 2: kalman
          controller: 2 # 1: PID, 2: mellinger
        ring:
          effect: 16 # 6: double spinner, 7: solid color, 16: packetRate
          solidBlue: 255 # if set to solid color
          solidGreen: 0 # if set to solid color
          solidRed: 0 # if set to solid color
          headlightEnable: 0
        locSrv:
          extPosStdDev: 1e-3
          extQuatStdDev: 0.5e-1
        kalman:
          resetEstimation: 1
      # tracking
      motion_capture_type: "optitrack" # one of none,vicon,optitrack,optitrack_closed_source,qualisys,vrpn
      object_tracking_type: "libobjecttracker" # one of motionCapture,libobjecttracker
      send_position_only: False # set to False to send position+orientation; set to True to send position only
      motion_capture_host_name: "192.168.0.100"
      # motion_capture_interface_ip: "" # optional for optitrack with multiple interfaces
      save_point_clouds: "/dev/null" # set to a valid path to log mocap point cloud binary file.
      print_latency: False
      write_csvs: False
      force_no_cache: False
      enable_parameters: True
      enable_logging: True
      enable_logging_pose: True
    </rosparam>
  </node>


  <node pkg="crazyswarm" type="crazyswarm_teleop" name="crazyswarm_teleop" output="screen">
    <param name="csv_file" value="$(find crazyswarm)/launch/figure8_smooth.csv" />
    <param name="timescale" value="0.8" />
  </node>

  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find crazyswarm)/launch/test.rviz"/>

  <!-- <node pkg="rqt_plot" type="rqt_plot" name="rqt_plot_x" args="/cf2/log1/values[0]"/> -->
  <!-- <node pkg="rqt_plot" type="rqt_plot" name="rqt_plot_roll" args="/cf1/log1/values[2] /cf1/log1/values[3]"/> -->

</launch>

```

**Comment by noahra** [comment]
I'm still having the same problem.. the drone flies for a while, then it starts to spin uncontrollably, it's just like the same motion when you flush something down a toilet.

**Reply by knmcguire** [reply]
Hi! have you tried a different object tracking type as well? See this documentation for an explanation for that: https://crazyswarm.readthedocs.io/en/latest/configuration.html#select-object-tracking-mode-motion-capture-only

Currently your tracking type relies on IMU for the attitude estimate, so there seems to be something off with the IMU of your crazyflie perhaps...

**Reply by noahra** [reply]
I tried a different drone and still, I am experiencing the same issue. Can the lack of a Flow deck be causing this? I also get the message:` Latency is too high` when I do the command "roslaunch crazyswarm hover_swarm.launch"

**Reply by whoenig** [reply]
You shouldn't need a flow deck, as long as you do motions in x/y (otherwise the Kalman filter cannot estimate yaw correctly). If you get the Latency is too high warning just once, it's nothing to worry about.

**Reply by noahra** [reply]
OK! I just get it once after I launch the command. The drone is shaky in general. And I have noticed that in the Motive software I am using, it is showing two markers instead of 1 (the one on the drone). Technically, it's blinking from 1 to 2 markers. I think the Optitrack Mocap system needs a recalibration....

**Comment by noahra** [comment]
Turns out it is the Crazyswarm goTo() function that makes the drone unstable after a while. I will have to figure out how to use cmdPosition instead.

**Comment by whoenig** [comment]
goTo() can be unstable if the provided duration is small (<0.5s). I am not sure why, but I assume that somehow the optimization problem solved online becomes numerically unstable.

**Comment by noahra** [comment]
I still have the same issue...

However, I found this old thread which describes my issue:

https://forum.bitcraze.io/viewtopic.php?t=4079

I am trying to use land() and takeoff() with the so called yaw-parameter set to "None" but that parameter does not seem to exist...

**Reply by whoenig** [reply]
What you describe/link to is not related to Crazyswarm, which has its own API.

**Reply by noahra** [reply]
> What you describe/link to is not related to Crazyswarm, which has its own API.

Oh you are right.. I must have been reading it too fast
